firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/crcelements.txt

```diff
@@ -0,0 +1,104 @@
+/* compute_body_crc
+Scenario: CRCElement: The compute_body_crc function correctly computes crc from body
+  GIVEN("A CRC element initialized with an empty payload")
+    WHEN("the crc is computed from the payload")
+      THEN("the computed crc is equal to the expected crc")
+
+  GIVEN("A CRC Element initialized with the payload '123456789'")
+    WHEN("the crc is computed from the payload")
+      THEN("the computed crc is equal to the expected crc")
+
+  GIVEN("A crc element is initalised with payload 0x00")
+    WHEN("the crc is computed from the payload")
+      THEN("the computed crc is equal to the expected crc")
+
+  GIVEN("A crc element is initalised with payload 0x01")
+    WHEN("the crc is computed from the payload")
+      THEN("the computed crc is equal to the expected crc")
+
+/* write function
+Scenario: CRCElement: The write function correctly generates body from computed crc and payload
```

Comment:
> Should we test the write method using a ConstructedCRCElement? This way we wouldn't need to add THENs asserting that the payload given in the constructor remains unchanged after a write, because the compiler enforces constness.

Yes, updated the tests for  `write` by using ConstructedCRCElement.